src/session/index.ts

```diff
@@ -85,68 +71,72 @@ export async function startSession(
     )
   }
 
-  const markdownOrNotebook = vscode.workspace
-    .getConfiguration('ai-task')
-    .get<string>('markdownOrNotebook')!
-
-  const {
-    // sessionHighLevelFeedbackDocument,
-    sessionMarkdownLowLevelFeedbackDocument,
-  } = await createSessionLogDocuments(markdownOrNotebook)
+  const { sessionMarkdownLowLevelFeedbackDocument } =
+    await createSessionLogDocuments()
 
   const lowLevelLogger = (text: string) =>
     queueAnAppendToDocument(sessionMarkdownLowLevelFeedbackDocument, text)
 
-  const highLevelLogger = async (text: string) => {
-    /*
-     * Always append to exeuction output
-     * taskAppendWithoutErasing(execution, text)
-     */
-    void queueAnAppendToExecutionOutput(execution, text)
+  const pendingEdits = new Map<string, Promise<void>>()
 
-    /*
-     * return markdownOrNotebook === 'markdown'
-     *   ? queueAnAppendToDocument(sessionHighLevelFeedbackDocument, text)
-     *   : queueAnAppendToMarkdownValue(sessionHighLevelFeedbackDocument, text)
-     */
-  }
+  const currentCellOutputContentMap = new Map<string, string>()
+  const cellEditQueue: Record<string, string[]> = {}
 
-  const cachedActiveEditor = vscode.window.activeTextEditor
+  async function queueAnAppendToExecutionOutput(
+    execution: vscode.NotebookCellExecution,
+    text: string,
+  ) {
+    if (execution.token.isCancellationRequested) {
```

Comment:
This looks like a much cleaner implementation from what we had before!
I have not followed the whole algorithm closely but looks correct from skimming it